src/censure.jl

```diff
@@ -0,0 +1,236 @@
+    	in_sum += A + D - B - C
+
+    	trap_top_right = I + CartesianIndex(- m_in2 - OF.n_in - 1, m_in2 - 1)
+    	trap_top_left = I + CartesianIndex(- m_in2 - OF.n_in - 1, - m_in2)
+    	trap_bot_right = I + CartesianIndex(- m_in2 - 1, m_in2 + OF.n_in - 1)
+    	trap_bot_left = I + CartesianIndex(- m_in2 - 1, - m_in2 - OF.n_in)
+    	A = trap_top_left >= CartesianIndex(1, 1) ? rs_img[trap_top_left] : 0.0
+    	B = trap_top_right >= CartesianIndex(1, 1) ? ls_img[trap_top_right] : 0.0
+    	C = trap_bot_left >= CartesianIndex(1, 1) ? rs_img[trap_bot_left] : 0.0
+    	D = trap_bot_right >= CartesianIndex(1, 1) ? ls_img[trap_bot_right] : 0.0
+    	in_sum += A + D - B - C
+
+    	topleft = I + CartesianIndex(- m_out2 - 1, - m_out2 - OF.n_out - 1)
+    	topright = I + CartesianIndex(m_out2, - m_out2 - OF.n_out - 1)
+    	bottomleft = I + CartesianIndex(- m_out2 - 1, m_out2 + OF.n_out)
+    	bottomright = I + CartesianIndex(m_out2, m_out2 + OF.n_out)
+    	A = topleft >= CartesianIndex(1, 1) ? int_img[topleft] : 0.0
+    	B = topright >= CartesianIndex(1, 1) ? int_img[topright] : 0.0
+    	C = bottomleft >= CartesianIndex(1, 1) ? int_img[bottomleft] : 0.0
+    	D = bottomright >= CartesianIndex(1, 1) ? int_img[bottomright] : 0.0
+    	out_sum = A + D - B - C
+
+    	trap_top_right = bottomright
+    	trap_bot_right = I + CartesianIndex(m_out2 + OF.n_out, m_out2)
+    	trap_top_left = I + CartesianIndex(m_out2, - m_out2 - OF.n_out)
+    	trap_bot_left = I + CartesianIndex(m_out2 + OF.n_out, - m_out2 - 1)
+    	A = trap_top_left >= CartesianIndex(1, 1) ? ls_img[trap_top_left] : 0.0
+    	B = trap_top_right >= CartesianIndex(1, 1) ? rs_img[trap_top_right] : 0.0
+    	C = trap_bot_left >= CartesianIndex(1, 1) ? ls_img[trap_bot_left] : 0.0
+    	D = trap_bot_right >= CartesianIndex(1, 1) ? rs_img[trap_bot_right] : 0.0
+    	out_sum += A + D - B - C
+
+    	trap_top_right = I + CartesianIndex(- m_out2 - OF.n_out - 1, m_out2 - 1)
+    	trap_top_left = I + CartesianIndex(- m_out2 - OF.n_out - 1, - m_out2)
+    	trap_bot_right = I + CartesianIndex(- m_out2 - 1, m_out2 + OF.n_out - 1)
+    	trap_bot_left = I + CartesianIndex(- m_out2 - 1, - m_out2 - OF.n_out)
+    	A = trap_top_left >= CartesianIndex(1, 1) ? rs_img[trap_top_left] : 0.0
+    	B = trap_top_right >= CartesianIndex(1, 1) ? ls_img[trap_top_right] : 0.0
+    	C = trap_bot_left >= CartesianIndex(1, 1) ? rs_img[trap_bot_left] : 0.0
+    	D = trap_bot_right >= CartesianIndex(1, 1) ? ls_img[trap_bot_right] : 0.0
+    	out_sum += A + D - B - C
+    	out_sum = out_sum - in_sum
+
+    	response[I] = in_sum * filter.in_weight - out_sum * filer.out_weight
+    end
+	response
+	
+end
+
+function checkFeature()
+end
+
+CENSURE(; smallest::Integer = 1, largest::Integer = 7, filter::Type = BoxFilter, responseThreshold::Number = 0.15, lineThreshold::Number = 10) = CENSURE(smallest, largest, filter, getFilterStack(filter, smallest, largest), responseThreshold, lineThreshold)
```

Comment:
BTW, can you wrap long lines? It means I have to scroll my browser window to find the "comment" button :smile:.